### ISSUE TITLE
SuperCircuit: Add PublicInputsCircuit

### ISSUE THREAD

**Issue by ed255** [privacy-scaling-explorations/zkevm-circuits#663]

**Comment by aguzmant103** [comment]
Comments from David: will focus on the spec first and after the Halo2 upgrade is done, retake this task.

Comment from CPerezz: don't think Halo2 upgrade will change the full codebase but will still to work out on some merge conflicts.

**Comment by aguzmant103** [comment]
Immediate task: fix the integration challenges (new task), then putting the public input circuit, then the spec will be almost written.

**Comment by aguzmant103** [comment]
Depends on  #747
